Source/DafnyLanguageServer/Workspace/DiagnosticPublisher.cs

```diff
@@ -70,10 +70,10 @@ public class DiagnosticPublisher : IDiagnosticPublisher { // TODO rename to Noti
         Version = document.Version,
         Diagnostics = document.GhostDiagnostics.ToArray(),
       };
-      if (previouslyPublishedGhostDiagnostics.TryGetValue(document.Uri, out var previousParams) && previousParams.Diagnostics.Equals(newParams.Diagnostics)) {
+      if (previouslyPublishedGhostDiagnostics.TryGetValue(document.Uri, out var previousParams) && previousParams.Diagnostics.SequenceEqual(newParams.Diagnostics)) {
```

Comment:
Woops, good catch.

Reply:
Well this wasn't needed for some reason, the tests already showed it was working, but this seems safer.